src/components/FormField/FormField.tsx

```diff
@@ -0,0 +1,141 @@
+    helperTextColor?: Validation;
+    hiddenLabel?: boolean;
+    readOnly?: boolean;
+    status?: Validation;
+    'data-test-id'?: string;
+};
+
+export const FormField = ({
+    children,
+    disabled = false,
+    error = false,
+    errorText,
+    errorTextColor = Validation.Error,
+    helperText,
+    helperTextColor = Validation.Default,
+    hiddenLabel = false,
+    label,
+    readOnly = false,
+    status = Validation.Default,
+    'data-test-id': dataTestId = 'fondue-form-field',
+}: FormFieldProps): ReactElement => {
+    const getFormattedLabel = (): ReactElement => {
+        const { text, required, secondaryLabel, hugWidth, tooltips } = label;
+        const elements: ReactElement[] = [];
+        const secondaryLabelStyle = 'tw-w-full tw-flex tw-flex-nowrap tw-justify-between';
+        const applySecondaryLabel = !hugWidth && secondaryLabel;
+        let formattedLabel = text;
+
+        if (required) {
+            formattedLabel += ' *';
+        }
+        if (tooltips?.length) {
+            const tooltipElements: ReactElement[] = tooltips.map((tip: TooltipProps) => (
+                <Tooltip
+                    key={`form-field-tooltip-${generateRandomId()}`}
+                    content={tip.content}
+                    size={tip.size}
+                    data-test-id={tip['data-test-id']}
+                >
+                    {tip.children}
+                </Tooltip>
+            ));
+            elements.push(
+                <span className={'tw-flex tw-gap-2'}>
+                    {formattedLabel}
+                    {...tooltipElements}
+                </span>,
+            );
+        } else {
+            elements.push(<span>{formattedLabel}</span>);
+        }
+
+        if (applySecondaryLabel) {
+            elements.push(<span>{secondaryLabel}</span>);
+        }
+        return (
+            <span
+                className={merge([
+                    'tw-whitespace-nowrap',
+                    hugWidth && 'tw-pt-2',
+                    applySecondaryLabel && secondaryLabelStyle,
+                    disabled || readOnly ? 'tw-text-black-60' : 'tw-text-black-80',
+                ])}
+                data-test-id={`${dataTestId}-label-text`}
+                aria-label={formattedLabel}
+            >
+                {...elements}
+            </span>
+        );
+    };
+
+    return (
+        <label
+            className={merge([
+                'tw-flex tw-text-text-weak tw-gap-2 tw-select-none tw-max-w-full',
+                label.hugWidth ? 'tw-w-fit tw-items-start tw-justify-start' : 'tw-w-full tw-flex-col',
+                disabled || readOnly
+                    ? 'hover:tw-cursor-not-allowed tw-pointer-events-none'
+                    : 'hover:tw-cursor-pointer hover:tw-text-text group-hover:tw-text-text',
+            ])}
+            data-test-id={`${dataTestId}-label`}
+            aria-label={label.text}
+        >
+            {hiddenLabel ? null : getFormattedLabel()}
```

Comment:
```suggestion
            {hiddenLabel ? null : <FormattedLabel />}
```

Reply:
@bojangles-m the issue I see with abstracting this method to a component is that is it very `prop` dependant and would be a bit "messy" to move it outside the main render.

Reply:
YOu dont need to extract it, just change the name :)

Reply:
SonarLint does not like this approach (and i agree).  
```Do not define components during render. React will see a new component type on every render and destroy the entire subtree\u8217s DOM nodes and state. Instead, move this component definition out of the parent component \u8220FormField\u8221 and pass data as props.```

Reply:
Made the change to a component. 👍

Reply:
great